src/components/Dropdown.tsx

```diff
@@ -185,51 +189,23 @@ class Box extends React.Component<any, any> {
     }
 }
 Box.displayName = 'Dropdown.Box'
-Box.propTypes = {
-    /** Whether the dropdown should open to the top. */
-    top: PropTypes.bool,
-    /** Whether the dropdown should open to the right. */
-    right: PropTypes.bool,
-    /** Id of the scrolling parent element to place dropdown in it. */
-    //eslint-disable-next-line @typescript-eslint/camelcase
-    scrolling_parent: PropTypes.string,
-    /** Whether to keep dropdown open when interacted with the Body content. */
-    allowBodyInteractions: PropTypes.bool,
-    /** Callback function when the body is shown. */
-    onShowBody: PropTypes.func,
-    /** Callback function when the body is hidden. */
-    onHideBody: PropTypes.func,
-    /** Additional css class applied to the Dropdown. */
-    className: PropTypes.string,
-    /** Should be two elements: Dropdown.Trigger and Dropdown.Body.
-     * Second element can be a function, which will be called only if it is open */
-    children: PropTypes.any,
-}
 
 interface TriggerProps {
     onClick?: (event?: React.MouseEvent) => void
 }
 
 class Trigger extends React.Component<TriggerProps> {
-    public static displayName
-    public static propTypes
-
-    /**
-     * @param {TriggerProps} props
-     * @param {unknown} context
-     */
-    constructor(props, context) {
+    public static displayName: string
+
+    constructor(props: TriggerProps, context: React.Context<any>) {
         super(props, context)
         this._onClick = this._onClick.bind(this)
     }
 
-    /**
-     * @param {React.MouseEvent} event
-     */
-    _onClick(event) {
+    _onClick(event: React.MouseEvent) {
         event.preventDefault()
         event.stopPropagation()
-        this.props.onClick(event)
+        this.props.onClick && this.props.onClick(event)
```

Comment:
I think it is more readable to use regular `if`s.

Reply:
Changed!